src/move_util.c

```diff
@@ -1,9 +1,25 @@
 #include "global.h"
 #include "move_util.h"
 
+#include "constants/ability.h"
 #include "constants/move_id.h"
 #include "constants/status.h"
 #include "moves.h"
+#include "dungeon_util.h"
+#include "dungeon_pokemon_attributes.h"
+#include "dungeon_random.h"
+
+bool8 sub_8057308(struct DungeonEntity *pokemon, s32 chance)
```

Comment:
This calculates if a secondary move effect will occur, like the burn chance on Flamethrower.